src/gConsent/manage/revokeAccessGrant.ts

```diff
@@ -38,29 +39,23 @@ import { getBaseAccess } from "../util/getBaseAccessVerifiableCredential";
 const { quad, namedNode } = DataFactory;
 
 /**
- * Makes a request to the access server to revoke a given Verifiable Credential (VC).
- *
- * @param vc Either a VC, or a URL to a VC, to be revoked.
- * @param options Optional properties to customise the request behaviour.
- * @returns A void promise.
- * @since 0.4.0
+ * @internal
  */
-async function revokeAccessGrant(
+async function revokeAccessCredential(
   vc: DatasetWithId | VerifiableCredential | URL | UrlString,
   options: Omit<AccessBaseOptions, "accessEndpoint"> = {},
+  type: NamedNode<string> = solidVc.SolidAccessGrant,
 ): Promise<void> {
-  const credential = await getBaseAccess(vc, options, solidVc.SolidAccessGrant);
+  const credential = await getBaseAccess(vc, options, type);
 
   if (
-    !credential.has(
-      quad(namedNode(getId(credential)), TYPE, solidVc.SolidAccessGrant),
-    ) &&
+    !credential.has(quad(namedNode(getId(credential)), TYPE, type)) &&
     !credential.has(
       quad(namedNode(getId(credential)), TYPE, solidVc.SolidAccessDenial),
     )
   ) {
     throw new Error(
-      `An error occurred when type checking the VC: Not of type [${solidVc.SolidAccessGrant.value}] or [${solidVc.SolidAccessDenial.value}].`,
+      `An error occurred when type checking the VC: Not of type [${type.value}] or [${solidVc.SolidAccessDenial.value}].`,
```

Comment:
```suggestion
      `An error occurred when type checking the VC: Not of type [${type.value}].`,
```
An Access Denial would not be accepted in all cases, so this error message should be specific to the provided VC type. However, this means that this is a breaking change: `revokeAccessGrant` supported both approved or denied Access Grants before this change, and now only approved ones.

You could slightly change the internal function so that it accepts a list of types, therefore potentially accepting more than one value.